pyccl/halos/profiles.py

```diff
@@ -167,6 +167,10 @@ def real(self, cosmo, r, M, a, mass_def=None):
             are scalars, the corresponding dimension will be
             squeezed out on output.
         """
+        # mass_def is irrelevant in these profiles
+        if self.name in ['Gaussian', 'PowerLaw']:
+            mass_def = None
+
```

Comment:
As I said in my top comment, just remove these, it's fine.

Reply:
done